src/main/scala/uk/gov/hmrc/clamav/ClamAntiVirus.scala

```diff
@@ -16,88 +16,51 @@
 
 package uk.gov.hmrc.clamav
 
-import java.io._
+import javax.inject.Inject
 
 import play.api.Logger
-import uk.gov.hmrc.clamav.ClamAntiVirus._
-import uk.gov.hmrc.clamav.config.ClamAvConfig
 
 import scala.concurrent.{ExecutionContext, Future}
-import scala.util.{Failure, Success, Try}
+import scala.util.Try
 
-trait ClamAvResponseInterpreter {
 
-  def interpretResponseFromClamd: PartialFunction[String, Try[Boolean]] = {
-    case `okClamAvResponse` =>
-      Logger.info("File clean")
-      Success(true)
-    case responseString =>
-      Logger.warn(s"Virus detected : $responseString")
-      Failure(new VirusDetectedException(responseString))
-  }
-
-  def invaldResponse = {
-    Logger.warn("Empty response from clamd")
-    Failure(new VirusScannerFailureException("Empty response from clamd"))
-  }
-}
-
-trait ClamAvSocket {
-  val config: ClamAvConfig
-
-  lazy val socket = openSocket()
+class ClamAntiVirus @Inject()(responseInterpreter: ClamAvResponseInterpreter,
+                              clamAvSocket: ClamAvSocket) {
```

Comment:
Hmm - do we use only one connection to ClamAV per whole service?
How do we deal with concurrent requests and what happens when connection terminates with exception?